fixed-hash/src/hash.rs

```diff
@@ -588,6 +588,7 @@ macro_rules! impl_rustc_hex_for_fixed_hash {
 			/// - When encountering invalid non hex-digits
 			/// - Upon empty string input or invalid input length in general
 			fn from_str(input: &str) -> $crate::core_::result::Result<$name, $crate::rustc_hex::FromHexError> {
+				let input = if input.starts_with("0x") { &input[2..] } else { input };
```

Comment:
alternatively we could use
```suggestion
				let input = input.strip_prefix("0x").unwrap_or(input);
```